eli5/keras/explain_prediction.py

```diff
@@ -167,68 +248,194 @@ def explain_prediction_keras_image(model,
+    if layer is not None:
+        activation_layer = _get_layer(model, layer)
+    else:
+        activation_layer = _autoget_layer_image(model)
+
+    vals = gradcam_backend_keras(model, doc, targets, activation_layer)
+    activations, grads, predicted_idx, predicted_val = vals
+    heatmap = gradcam_heatmap(activations,
+                              grads,
+                              relu=relu,
+                              counterfactual=counterfactual,
+                              )
+    # take from batch
+    predicted_idx, = predicted_idx
+    predicted_val, = predicted_val
+    heatmap, = heatmap
 
     return Explanation(
         model.name,
-        description=DESCRIPTION_KERAS,
+        description=DESCRIPTION_GRADCAM,
         error='',
         method='Grad-CAM',
-        image=image,
+        image=image, # RGBA Pillow image
+        layer=activation_layer.name,
         targets=[TargetExplanation(
             predicted_idx,
             score=predicted_val, # for now we keep the prediction in the .score field (not .proba)
-            heatmap=heatmap, # 2D [0, 1] numpy array
+            heatmap=heatmap, # 2D numpy array
         )],
         is_regression=False, # might be relevant later when explaining for regression tasks
-        highlight_spaces=None, # might be relevant later when explaining text models
     )
 
 
+def explain_prediction_keras_text(model,
+                                  doc,
+                                  tokens=None, # type: Optional[Union[List[str], np.ndarray]]
+                                  pad_value=None, # type: Optional[Union[int, float]]
+                                  pad_token=None, # type: Optional[str]
+                                  interpolation_kind='linear', # type: Union[str, int]
+                                  targets=None,
+                                  layer=None,
+                                  relu=True,
+                                  counterfactual=False,
+                                  ):
+    """
+    Explain a text-based model, highlighting parts of text that contributed to the prediction.
+
+    In the case of binary classification, this highlights what makes the output go up.
+
+    See :func:`eli5.keras.explain_prediction.explain_prediction_keras` for description of ``targets``, 
+    ``layer``, ``relu``, and ``counterfactual`` parameters.
+
+    :param numpy.ndarray doc:
+        Suitable input tensor. Temporal with batch size. May have padding.
+
+    :param tokens:
+        Tokens that correspond to ``doc``.
+        With padding if ``doc`` has padding.
+
+        A Python list or a numpy array of strings. With the same length as ``doc``.
+        If ``doc`` has batch size = 1, batch dimension from tokens may be omitted.
+
+        These tokens will be highlighted for text-based explanations.
+    :type tokens: list[str], optional
+
+    :param pad_value:
+        Number for padding. If given, cuts padding off.
```

Comment:
Probably a more common way to say this would be something like
```suggestion
        Integer identified of the padding token. If given, cuts padding off.
```